applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -549,37 +553,42 @@ static bool key_value_set(struct items *items, u16_t usage_id, s16_t value)
 	return update_needed;
 }
 
-static void send_report_keyboard(void)
+static void send_report_keyboard(u8_t report_id, struct report_data *rd)
 {
 	if (!IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_KEYBOARD_SUPPORT)) {
 		/* Not supported. */
 		__ASSERT_NO_MSG(false);
 		return;
 	}
 
-	struct report_data *rd = &state.report_data[IN_REPORT_KEYBOARD_KEYS];
+	BUILD_ASSERT_MSG(REPORT_SIZE_KEYBOARD_KEYS == KEYBOARD_REPORT_KEY_COUNT_MAX + 2,
+			 "Incorrect keyboard report size");
 
-	const size_t max = ARRAY_SIZE(rd->items.item);
+	/* Encode report. */
 
-	struct hid_keyboard_event *event = new_hid_keyboard_event();
+	struct hid_report_event *event = new_hid_report_event(sizeof(report_id) + REPORT_SIZE_KEYBOARD_KEYS);
 
 	event->subscriber = state.selected->id;
-	event->modifier_bm = 0;
 
+	event->dyndata.data[0] = report_id;
```

Comment:
Maybe we could assert that report ID is keyboard keys report ID.

Reply:
Also for other reports.

Reply:
defensive but why not. it should be optimized out

Reply:
done